sw/device/silicon_creator/rom_ext/bootstrap.c

```diff
@@ -99,5 +99,48 @@ rom_error_t rom_ext_bootstrap(void) {
   }
   HARDENED_CHECK_EQ(enabled, kHardenedBoolTrue);
 
+  enum {
+    kNumPagesInRomExt = CHIP_ROM_EXT_SIZE_MAX / FLASH_CTRL_PARAM_BYTES_PER_PAGE
+  };
+
+  static_assert(kNumPagesInRomExt <= FLASH_CTRL_PARAM_REG_PAGES_PER_BANK,
+                "ROM_EXT must fit within a single flash bank.");
+
+  // The region of slot A following the ROM_EXT region is erasable and
+  // programmable.
+  flash_ctrl_data_region_protect(
```

Comment:
just to confirm: bootstrap requires read access to verify erases, right?

Reply:
Yes, I believe that's true, since [flash_ctrl_data_erase_verify()](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/lib/drivers/flash_ctrl.c;l=451-456;drc=1ff6f6ea02a1711224a284a8855264406750eabd) is defined in terms of abs_mmio_read32().

Thanks for pointing that out — I just added a check to the FlashCtrlSim class in the ROM_EXT bootstrap tests.